### PR TITLE
Rename Preferences endpoints

### PR DESCRIPTION
**Pull request by gokulavasan**
JIRA: https://issues.cask.co/browse/CDAP-1654

Bamboo: http://builds.cask.co/browse/CDAP-DUT934

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
This does not change the APIS as per JIRA. 
For instance: We wanted to rename
`v3/configuration/preferences/namespaces/{id}` to `v3/namespaces/{id}/preferences` in the JIRA>

**Comment by gokulavasan** [comment]
@sreevatsanraman Updated PR. Please review.

**Comment by gokulavasan** [comment]
@ajainarayanan @gaarf This PR will change the endpoints used for Console Settings (user specific) in the UI. Please refer to the JIRA for the REST API change.

**Comment by gaarf** [comment]
@gokulavasan @sreevatsanraman does this mean per-user configuration is now also per-namespace?

**Comment by gokulavasan** [comment]
@sreevatsanraman Build passed. Please review.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/PreferencesHttpHandlerTest.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
  * Tests for {@link PreferencesHttpHandler}
  */
 public class PreferencesHttpHandlerTest extends AppFabricTestBase {
+
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/PreferencesHttpHandlerTest.java]: We don't have test for other preferences endpoints? 
* Namespace level
* App level
* Program level

Can you add them if they are not there currently?

Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/PreferencesHttpHandlerTest.java]: There are tests that test those endpoints. setProperty, getProperty, deleteProperty are helper methods that help construct the right URI.

**Comment by gokulavasan** [comment]
@sreevatsanraman Can you please review? I have addressed your comments.

**Comment by sreevatsanraman** [comment]
LGTM :+1:

**Pull request merged**